app/services/one_off_redirect_fix_service.rb

```diff
@@ -0,0 +1,48 @@
+class OneOffRedirectFixService
+  def self.fix_redirects!
+    editions = Edition.where(document_type: "redirect").joins(:document).where.not("documents.locale = 'en'")
+    editions_updated = 0
+    editions.each do |edition|
+      updated = update_redirects(edition)
+      if updated
+        Commands::V2::RepresentDownstream.new.call([edition.content_id], queue: DownstreamQueue::LOW_QUEUE)
+        editions_updated += 1
+      end
+    end
+    Rails.logger.info "#{editions_updated} editions have been updated."
+  end
+
+  def self.update_redirects(edition)
+    return false if edition.redirects.blank?
+
+    if edition.redirects.count > 1
+      Rails.logger.info "[RedirectDepthFix] edition #{edition.id} has more than one redirect."
+      return false
+    end
+
+    next_base_path = edition.redirects.first[:destination]
+
+    while (next_edition = Edition.where(base_path: next_base_path).order(:updated_at).last)
+      break unless next_edition && next_edition.redirects.present?
+
+      if next_base_path == edition.base_path
+        Rails.logger.info "[RedirectDepthFix] edition #{edition.id} and #{next_edition.id} has a circular redirect"
```

Comment:
Nitpick: in each `Rails.logger.info`, we have a `[RedirectDepthFix]` prefix. I think that prefix only really applies to this one (circular redirect == infinite depth), whereas the other two should probably be changed to something like `[RedirectsValueInvalid]`.